README.md

```diff
@@ -384,6 +384,7 @@ of falling back to your own custom handler.
 Usage is like so:
 
 ```rust
+#[cfg(feature = "fancy")]
```

Comment:
@LordMZTE do you still work on this one ?

Reply:
Well, I'm not sure how to continue with this, or if this is something we should even worry about.

Reply:
```suggestion
# #[cfg(feature = "fancy")]
```

Would hide the `cfg` attribute in the generated documentation, avoiding the copy/paste issue. Still results in the test being silently ignored when the `fancy` feature isn't enabled, which I'm not sure is desirable, since `set_hook` is available either way.

If we go the route of "tests don't work without --all-features" , it would be nice if there was a clearer indication of _why_ the tests are failing. When I first started looking at this project, I was definitely confused by the failing tests, and it took a bit to find the relevant part in `CONTRIBUTING.md` that explained it.

Something like this:

```rust
#[cfg(all(test, not(feature = "fancy")))]
compile_error!("Tests only work with the 'fancy' feature enabled. Try 'cargo test --features fancy'.");
```